platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/CrlChecker.java

```diff
@@ -154,4 +132,82 @@ Properties loadProperties(String location) {
         return PropertiesLoader.loadProperties(location);
     }
 
+    /**
+     * Runnable to refresh the CRL from the URL when either the nextUpdate time has elasped
+     * or it has rolled over to the next day.
+     */
+    private class CrlRefresh implements Runnable {
+
+        private Calendar start = Calendar.getInstance();
+
+        public void run() {
+            Properties encryptionProperties = loadProperties(encryptionPropertiesLocation);
+            CRL crl = crlCache.get();
+            if (crl instanceof X509CRL) {
+                X509CRL x509Crl = (X509CRL) crl;
+                if (Calendar.getInstance()
+                        .getTime()
+                        .after(x509Crl.getNextUpdate()) || rolledOverDay()) {
+                    setCrlLocation(encryptionProperties.getProperty(CRL_PROPERTY_KEY));
+                    start = Calendar.getInstance();
+                }
+            } else if (crl == null) {
+                setCrlLocation(encryptionProperties.getProperty(CRL_PROPERTY_KEY));
+                start = Calendar.getInstance();
+            }
+        }
+
+        /**
+         * Sets the location of the CRL. Enables CRL checking if property is set, disables it otherwise
+         *
+         * @param location Location of the DER-encoded CRL file that should be used to
+         *                 check certificate revocation.
+         */
+        public void setCrlLocation(String location) {
+            CRL current = crlCache.get();
+            if (location == null) {
+                LOGGER.info("CRL property in {} is not set. Certs will not be checked against a CRL",
+                        encryptionPropertiesLocation);
+                crlCache.compareAndSet(current, null);
+            } else {
+                CRL crl = createCrl(location);
+                if (crl != null) {
+                    crlCache.compareAndSet(current, crl);
+                }
+            }
+        }
+
+        /**
+         * Generates a new CRL object from the given location.
+         *
+         * @param location File Path or URL to the CRL file
+         * @return A CRL object constructed from the given file path or URL. Null if an error occurred while attempting to read the file.
+         */
+        private CRL createCrl(String location) {
+            URL url = urlFromPath(location);
+
+            //If we get a URL, use URL, otherwise use as local file path
+            try (InputStream is = url != null ? url.openStream() : new FileInputStream(new File(
+                    location))) {
+
+                CertificateFactory cf = CertificateFactory.getInstance("X.509");
+                return cf.generateCRL(is);
+            } catch (IOException e) {
+                LOGGER.debug("An error occurred while accessing {}", location, e);
```

Comment:
It looks like this exception could occur if the CRL location is invalid.  I'm thinking the condition should be logged at a higher level to bring this to the attention of the administrator.